store/v2/root/snapshot.go

```diff
@@ -0,0 +1,216 @@
+package root
+
+import (
+	"bufio"
+	"compress/zlib"
+	"fmt"
+	prefixdb "github.com/cosmos/cosmos-sdk/db/prefix"
+	"github.com/cosmos/cosmos-sdk/snapshots"
+	snapshottypes "github.com/cosmos/cosmos-sdk/snapshots/types"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	types "github.com/cosmos/cosmos-sdk/store/v2"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	protoio "github.com/gogo/protobuf/io"
+	"io"
+	"math"
+)
+
+// Restore implements snapshottypes.Snapshotter.
+func (rs *Store) Restore(height uint64, format uint32, chunks <-chan io.ReadCloser, ready chan<- struct{}) error {
+	if height == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrLogic, "cannot restore snapshot at height 0")
+	}
+	if height > uint64(math.MaxInt64) {
+		return sdkerrors.Wrapf(snapshottypes.ErrInvalidMetadata,
+			"snapshot height %v cannot exceed %v", height, int64(math.MaxInt64))
+	}
+
+	versions, err := rs.stateDB.Versions()
+	if versions.Count() != 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrLogic, "cannot restore snapshot %v", height)
+	}
+
+	// Signal readiness. Must be done before the readers below are set up, since the zlib
+	// reader reads from the stream on initialization, potentially causing deadlocks.
+	if ready != nil {
+		close(ready)
+	}
+
+	// Set up a restore stream pipeline
+	// chan io.ReadCloser -> chunkReader -> zlib -> delimited Protobuf -> ExportNode
+	chunkReader := snapshots.NewChunkReader(chunks)
+	defer chunkReader.Close()
+	zReader, err := zlib.NewReader(chunkReader)
+	if err != nil {
+		return sdkerrors.Wrap(err, "zlib failure")
+	}
+	defer zReader.Close()
+	protoReader := protoio.NewDelimitedReader(zReader, snapshotMaxItemSize)
+	defer protoReader.Close()
+
+	var subStore *substore
+	// initialisation empty store-schema for snapshot
+	ret := prefixRegistry{StoreSchema: StoreSchema{}}
```

Comment:
probably shouldn't call this `ret`, maybe `reg`

Reply:
Actually, we don't need to use prefixRegistry here, since we aren't checking for prefix conflicts. We can just use `StoreSchema` directly